metaflow/task.py

```diff
@@ -513,204 +519,239 @@ def run_step(
                 }
             }
         )
-        logger = self.event_logger
+
+        # 6. Update event logger with additional context around run_id, step_name, task_id, retry_count, and
+        # project_info to event logger context
+        # We don't update the context for the monitor as the cardinality of this context is high
+        event_logger_context = {
+            "run_id": run_id,
+            "step_name": step_name,
+            "task_id": task_id,
+            "retry_count": retry_count,
+            "project_name": current.get("project_name"),
+            "branch_name": current.get("branch_name"),
+            "is_user_branch": current.get("is_user_branch"),
+            "is_production": current.get("is_production"),
+            "project_flow_name": current.get("project_flow_name"),
+            "trace_id": trace_id or None,
+        }
+        self.event_logger.send(
```

Comment:
why use `send` instead of `log`? iirc `send` was a workaround for a very specific use case (I need to dig through the slack conversations) and we would want to avoid introducing it too much in the code base.

Reply:
`send` is used to update the `event_logger` context so that we have all these values available as additional tags in the event stream. We thus log this event with message type set to `MUST_SEND` instead of `BEST_EFFORT` that is normally used in `log`.

Reply:
Yes, the intent is to set a "common set of tags" (in the lingo of the other measure thing (datadog)). I think it may make sense to actually have a specific method `self.event_logger.set_common_tags()` or something like that. Backends would be free to do whatever they want with that. Internally, we could then use send to send a MUST_SEND message or just keep it around (the MUST_SEND was to avoid overloading the pipe to the sidecar with all the context all the time).

Reply:
Yes, lets discuss this in the call. What you said should be possible - depending on the presence of certain specific key/value pair in the payload we can use `MUST_SEND` or `BEST_EFFORT` appropriately. My only concern would be that if someone reads the code in `cli.py` then it would be confusing for them as to why we use different message types for different payloads.